packages/console/src/widget.ts

```diff
@@ -337,9 +337,14 @@ export class CodeConsole extends Widget {
    *
    * @returns A promise that indicates when the injected cell's execution ends.
    */
-  inject(code: string): Promise<void> {
+  inject(code: string, metadata: JSONObject = {}): Promise<void> {
     let cell = this.createCodeCell();
     cell.model.value.text = code;
+    if (metadata) {
+      for (let key in metadata) {
```

Comment:
FYI, in general iterating over an object using `in` can iterate over inherited properties you typically don't want to iterate over. More common patterns these days are to use `Object.keys(metadata).forEach(() => {cell.model.metadata.set(key, metadata[key]);});`, or even:

```typescript
for (let key of Object.keys(metadata)) {
         cell.model.metadata.set(key, metadata[key]); 
}
```